clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsClusterProvider.java

```diff
@@ -36,4 +46,46 @@ public EcsClusterProvider(Cache cacheView) {
   public Collection<EcsCluster> getAllEcsClusters() {
     return ecsClusterCacheClient.getAll();
   }
+
+  public Collection<Cluster> getAllEcsClustersDescription(String account, String region) {
+    List<String> clusterNames = new ArrayList<>();
+    List<Cluster> clusters = new ArrayList<>();
+    Collection<EcsCluster> ecsClusters = ecsClusterCacheClient.getAll();
+    AmazonECS client = getAmazonEcsClient(account, region);
+    if (ecsClusters.size() > 0 && ecsClusters != null) {
+      for (EcsCluster ecsCluster : ecsClusters) {
+        clusterNames.add(ecsCluster.getName());
+        if (clusterNames.size() % EcsClusterDescriptionMaxSize == 0) {
+          List<Cluster> describeClusterResponse = getDescribeClusters(client, clusterNames);
+          if (describeClusterResponse.size() > 0) {
+            clusters.addAll(describeClusterResponse);
+          }
+          clusterNames.clear();
+        }
+      }
+      if (clusterNames.size() % EcsClusterDescriptionMaxSize != 0) {
+        List<Cluster> describeClusterResponse = getDescribeClusters(client, clusterNames);
+        if (describeClusterResponse.size() > 0) {
+          clusters.addAll(describeClusterResponse);
+        }
+      }
```

Comment:
This logic is identical. To improve maintainability, we should avoid copying and pasting duplicate code. Instead it would be beneficial to move this logic into a separate method that can be called in both places.

Reply:
This comment still needs to be addressed.

Reply:
resolved in the latest commit, please take a look at it and let me know if that addresses your concern.